app/models/scenario.rb

```diff
@@ -119,7 +119,11 @@ def area
   end
 
   def scaled?
-    scaler.present? || area[:derived]
+    scaler.present? || derived?
+  end
+
+  def derived?
```

Comment:
I get where you're going with this, but it feels wrong to ask the scenario if *it* is derived. The scenario isn't derived; the area is. Preferably we could do `scenario.area.derived?` but that isn't possible on account of `area` being a hash (for reasons that escape me).

Perhaps this method could be called something like `area_derived?`?

It feels a *bit* unnecessary since it is called from only one place. It could instead be abstracted away in `Area`.

```ruby
class Area
  def self.derived?(area)
    area[:derived]
  end

  def self.derived_code?(area_code)
    Atlas::Dataset.exists?(area_code) && derived?(get(area_code))
  end

  # ...
end
```

 I have no strong preference either way.